src/rpc_connection.c

```diff
@@ -1287,33 +1349,22 @@ rpc_connection_create(void *cookie, rpc_object_t params)
 		return (NULL);
 	}
 
-	conn = g_malloc0(sizeof(*conn));
+	conn = rpc_connection_init();
+
 	conn->rco_client = client;
-	g_mutex_init(&conn->rco_mtx);
-	g_mutex_init(&conn->rco_ref_mtx);
-	g_mutex_init(&conn->rco_send_mtx);
-	g_mutex_init(&conn->rco_subscription_mtx);
-	g_rw_lock_init(&conn->rco_call_rwlock);
-	g_rw_lock_init(&conn->rco_icall_rwlock);
 	conn->rco_flags = transport->flags;
 	conn->rco_params = params;
 	conn->rco_uri = client->rci_uri;
 	conn->rco_main_context = rpc_client_get_main_context(client);
-	conn->rco_calls = g_hash_table_new(g_str_hash, g_str_equal);
-	conn->rco_inbound_calls = g_hash_table_new(g_str_hash, g_str_equal);
-	conn->rco_subscriptions = g_ptr_array_new();
-	conn->rco_rpc_timeout = DEFAULT_RPC_TIMEOUT;
-	conn->rco_recv_msg = rpc_recv_msg;
```

Comment:
I see these lines removed but they were not readded anywhere (I grepped for `rpc_recv_msg` or `DEFAULT_RPC_TIMEOUT`. What am I missing?

Reply:
They are in rpc_connection_init()